packages/concerto-core/lib/introspect/decorator.js

```diff
@@ -78,6 +79,10 @@ class Decorator {
                 if (thing) {
                     if (thing.$class === `${MetaModelNamespace}.DecoratorTypeReference`) {
                         // XXX Is this really what we want?
+                        if (!MetaModelNamespace[thing.type.name]) {
```

Comment:
Are you sure if the `MetaModelNamespace` has the list of all the declarations present within the namespace?

https://github.com/accordproject/concerto-metamodel/blob/main/index.js#L29
https://concerto.accordproject.org/docs/reference/extending-metamodel/

Reply:
What condition will this not throw an error?

Reply:
I've Looked at it and found it string, so it can't do that, I'm currently looking for a way to find whether the type in the metamodel or not.